runtime/src/accounts_db.rs

```diff
@@ -3243,6 +3243,147 @@ impl AccountsDb {
         total_accounts_after_shrink
     }
 
+    pub(crate) fn minimize_accounts_db(
+        &self,
+        snapshot_slot: Slot,
+        minimized_account_set: &HashSet<Pubkey>,
+    ) {
+        let minimized_slot_set = self.get_minimized_slot_set(minimized_account_set);
+
+        let mut dead_slots = Vec::new();
+        let mut dead_storages = Vec::new();
+        for storages in self.get_snapshot_storages(snapshot_slot, None, None).0 {
+            let slot = storages.first().unwrap().slot();
+            if slot == snapshot_slot {
+                continue;
+            }
+            if minimized_slot_set.contains(&slot) {
+                self.filter_storages(storages, minimized_account_set, &mut dead_storages);
+            } else {
+                dead_slots.push(slot);
+            }
+        }
+
+        self.purge_slots_from_cache_and_store(
+            dead_slots.iter(),
+            &self.external_purge_slots_stats,
+            true,
+        );
+        self.drop_or_recycle_stores(dead_storages);
+    }
+
+    fn get_minimized_slot_set(&self, minimized_account_set: &HashSet<Pubkey>) -> HashSet<Slot> {
+        let mut minimized_slot_set = HashSet::new();
+        self.thread_pool_clean.install(|| {
+            minimized_slot_set.par_extend(minimized_account_set.par_iter().flat_map(|pubkey| {
+                if let Some(read_entry) = self.accounts_index.get_account_read_entry(pubkey) {
+                    read_entry
+                        .slot_list()
+                        .iter()
+                        .map(|(slot, _)| *slot)
+                        .collect()
+                } else {
+                    HashSet::new()
+                }
+            }));
+        });
+        minimized_slot_set
+    }
+
+    fn filter_storages(
+        &self,
+        storages: Vec<Arc<AccountStorageEntry>>,
+        minimized_account_set: &HashSet<Pubkey>,
+        dead_storages: &mut Vec<Arc<AccountStorageEntry>>,
+    ) {
+        let slot = storages.first().unwrap().slot();
+        let (stored_accounts, _, _) = self.get_unique_accounts_from_storages(storages.iter());
+        let mut stored_accounts = stored_accounts.into_iter().collect::<Vec<_>>();
+        stored_accounts.sort_unstable_by(|a, b| a.0.cmp(&b.0));
```

Comment:
tuple should sort by element 0 first in default implementation, as long as rest of tuple has partial_eq or whatever.

Reply:
so this could maybe just be .sort_unstable()

Reply:
`FoundStoredAccount<'_>` doesn't have the cmp trait, so we need to specify the sorting is ONLY by pubkey.